iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -48,6 +48,87 @@ void addCPUVectorizationPassPipeline(OpPassManager &passManager,
+  }
+  // Verify that the translation info is using the right pipeline.
+  auto pipeline =
+      IREE::Codegen::DispatchLoweringPassPipeline::CPUTensorToVectors;
+  StringRef pipelineName = stringifyEnum(pipeline);
+  if (translationInfo.getDispatchLoweringPassPipeline() != pipeline) {
+    return op->emitOpError("expected pipeline in translation.info to be ")
+           << pipelineName;
+  }
+
+  // Verify that the workload per workgroup is set and is non-zero.
+  SmallVector<int64_t> workloadPerWorkgroup =
+      translationInfo.getWorkloadPerWorkgroupVals();
+  SmallVector<unsigned> partitionedLoops = getPartitionedLoops(op);
+  if (workloadPerWorkgroup.size() != partitionedLoops.size()) {
+    return op->emitOpError("expected ")
+           << partitionedLoops.size()
+           << " entries for workload_per_wg, but got "
+           << workloadPerWorkgroup.size();
+  }
+  if (llvm::any_of(workloadPerWorkgroup,
+                   [](int64_t val) { return val == 0; })) {
+    return op->emitOpError("invalid to use 0 in workload_per_wg");
+  }
+
+  if (loweringConfig.getTileSizes().size() != 3) {
+    return op->emitOpError("expected three levels of tile sizes for ")
+           << pipelineName << ", got " << loweringConfig.getTileSizes().size();
+  }
+  SmallVector<int64_t> firstLevelTileSizes = loweringConfig.getTileSizeVals(0);
+  if (!firstLevelTileSizes.empty()) {
+    // Verify that if the first-level tile sizes are set, they are the same as
+    // workload_per_wg for the partitioned loops.
+    size_t minElements =
+        (partitionedLoops.empty() ? 0 : partitionedLoops.back() + 1);
+    if (firstLevelTileSizes.size() < minElements) {
+      return op->emitOpError("expected at least ")
+             << minElements
+             << " size for first level tiling to get the distribution fully "
+                "specified.";
+    }
+    llvm::SmallDenseSet<unsigned> partitionedLoopsSet;
+    partitionedLoopsSet.insert(partitionedLoops.begin(),
+                               partitionedLoops.end());
+    SmallVector<int64_t> partitionedTileSizes;
+    for (auto tileSize : llvm::enumerate(firstLevelTileSizes)) {
+      if (!partitionedLoopsSet.count(tileSize.index())) {
+        continue;
+      }
+      partitionedTileSizes.push_back(tileSize.value());
+    }
+    for (auto val : llvm::enumerate(llvm::reverse(workloadPerWorkgroup))) {
+      if (val.value() != partitionedTileSizes[val.index()]) {
+        return op->emitOpError("mismatch in distributed tile size value ")
+               << partitionedTileSizes[val.index()] << " at position "
+               << val.index() << " and workload_per_wg value " << val.value();
+      }
+    }
+  }
+
+  // Verify that native vector size is either empty, or if set is same as the
+  // last level of tiling
+  SmallVector<int64_t> nativeVectorSize =
+      loweringConfig.getNativeVectorSizeVals();
+  if (!nativeVectorSize.empty()) {
```

Comment:
I think this is already checked by `LoweringConfigAttr::verify` itself?

https://github.com/google/iree/blob/dacf2b19798d2251409d78b0faf88d07756faa26/iree/compiler/Codegen/Dialect/LoweringConfig.cpp#L158-L175

Reply:
Thats the `tileSizes`. not the `nativeVectorSize`.

Reply:
It truncates what I'd like to show. If you scroll it down, you'll see this. But now I found that you're correct. You are checking if it is empty. This is just check that it should exist.

https://github.com/google/iree/blob/dacf2b19798d2251409d78b0faf88d07756faa26/iree/compiler/Codegen/Dialect/LoweringConfig.cpp#L172-L175